addon/object/index.js

```diff
@@ -151,22 +153,24 @@ export const computed = decoratorWithParams(function(target, key, desc, params)
     throw new Error(`ES6 property getters/setters only need to be decorated once, '${key}' was decorated on both the getter and the setter`);
   }
 
-  let value;
-
   if (desc.writable === undefined) {
-    value = {
-      get: desc.get,
-      set: desc.set
-    };
+    const { get, set } = desc;
 
     // Unset the getter and setter so the descriptor just has a plain value
     desc.get = undefined;
     desc.set = undefined;
+
+    // Use a standard ember computed since getter/setter arrity is restricted,
+    // meaning ember-macro-helpers doesn't provide any benefit
+    return emberComputed(...params, {
+      get,
+      set(key, value) {
+        set(value);
```

Comment:
I think this needs to use some sort of binding to ensure correct `this` (e.g. `.call` / `.apply`).